### ISSUE TITLE
Python API : Validation after every few epochs

### ISSUE THREAD

**Issue by ujjwal-researcher** [Microsoft/CNTK#2137]
Given a CNN architecture, it is advisable to evaluate the model performance on the validation set after every few epochs (let's say 1). So, in this case the model is trained for an epoch and then validated over the validation set. Then it is again trained on the training set for another epoch and so on and so forth.

This helps in detecting signs of overtraining of the network.

Unfortunately, to the best of my knowledge a clear example or documentation is not available in CNTK which can help in achieving this. This seems to be a major drawback.

I sincerely request that some information be added on this matter as soon as possible.

**Comment by KeDengMS** [comment]
Thanks for the suggestion. Please check this [tutorial](https://github.com/Microsoft/CNTK/blob/master/Tutorials/CNTK_200_GuidedTour.ipynb) for cross-validation based training control. You can also refer to the [CrossValidationConfig](https://www.cntk.ai/pythondocs/cntk.train.training_session.html?highlight=crossvalidationconfig#cntk.train.training_session.CrossValidationConfig) API if you are using TrainingSession.

**Comment by NO-0044** [comment]
I tried to use CrossValidationConfig in trainin_session with distributed learner. I follow the tutorial above and the example of parallel training. However, seems cross validation doesn't work in this situation. There is no information printed and even I set the call back to direct return False won't stop training. However,just changing the distributed learner to local learner which disable parallel training can make cross validation work again. Is there any extra setting should make?

**Comment by KeDengMS** [comment]
It might have something to do frequency. Does it not work even with frequency == 1?

**Comment by NO-0044** [comment]
Yes, even with frequency = 1 the cv doesn't get executed

**Comment by KeDengMS** [comment]
Can you share a repro for this? Seems like a bug.
